workspaces/cli-server/src/routers/capture-router.ts

```diff
@@ -135,7 +135,31 @@ export function makeRouter(dependencies: ICaptureRouterDependencies) {
     }
 
     let diffsStream = diffQueries.diffs();
-    toJSONArray(diffsStream).pipe(res).type('application/json');
+
+    async function* normalizeDiffs(resultsStream: Readable) {
+      let pointersByFingerprint: Map<String, string[]> = new Map();
+      let diffs: [any, string][] = [];
+
+      for await (let [diff, pointers, fingerprint] of resultsStream) {
+        let existingPointers = pointersByFingerprint.get(fingerprint) || [];
+        if (existingPointers.length < 1) {
+          diffs.push([diff, fingerprint]);
+        }
+        pointersByFingerprint.set(
+          fingerprint,
+          existingPointers.concat(pointers)
+        );
+      }
```

Comment:
Just realised this is going to fail against the non-Rust version of the diff engine, which doesn't include the fingerprint with each result. We can either move this normalisation to the `DiffQueries.diffs` implementation, where a fork already exists, or we can adapt this to yield the result untouched when there's no fingerprint. Not sure which has my preference yet, or if it even matters which one we choose.

Reply:
Of course, I ended up doing both: https://github.com/opticdev/optic/pull/388/commits/cc43c8408c0b430251b5ecf5f4e8af7c1d08088c.